PhysicsTools/PatAlgos/interface/MuonMvaIDEstimator.h

```diff
@@ -0,0 +1,37 @@
+#ifndef __PhysicsTools_PatAlgos_MuonMvaIDEstimator__
+#define __PhysicsTools_PatAlgos_MuonMvaIDEstimator__
+
+#include <memory>
+#include <string>
+#include "PhysicsTools/ONNXRuntime/interface/ONNXRuntime.h"
+#include "FWCore/Framework/interface/stream/EDProducer.h"
+#include "FWCore/Utilities/interface/StreamID.h"
+#include "FWCore/Framework/interface/stream/EDAnalyzer.h"
+//
+
+namespace pat {
+  class Muon;
+}
+
+namespace edm {
+  class FileInPath;
+}
+
+namespace pat {
+  class MuonMvaIDEstimator {
+  public:
+    MuonMvaIDEstimator(const edm::FileInPath &weightsfile);
+    ~MuonMvaIDEstimator();
+
+    static void fillDescriptions(edm::ConfigurationDescriptions &);
+    static std::unique_ptr<cms::Ort::ONNXRuntime> initializeGlobalCache(const edm::ParameterSet &);
+    static void globalEndJob(const cms::Ort::ONNXRuntime *);
+    std::vector<float> computeMVAID(const pat::Muon &imuon) const;
```

Comment:
can you return a fixed-size array (`std::array<float, N>`) here, or does the model output size vary?

Reply:
The size does not vary, but the ONNXRuntime::run  output is a `vector<float>`, so it can not be defined as `std::array<float, N>`.